docs/en/sql-reference/statements/select/with.md

```diff
@@ -4,13 +4,13 @@ toc_title: WITH
 
 # WITH Clause {#with-clause}
 
-This section provides support for Common Table Expressions ([CTE](https://en.wikipedia.org/wiki/Hierarchical_and_recursive_queries_in_SQL)), so the results of `WITH` clause can be used in the rest of `SELECT` query.
+This section provides support for Common Table Expressions ([CTE](https://en.wikipedia.org/wiki/Hierarchical_and_recursive_queries_in_SQL)), so the results of `WITH` clause can be used in the rest of `SELECT` query. Named subqueries can be included to the current and children query context. Those identifiers can appear in places where table objects are allowed. Recursion is prevented by hiding the current level CTEs from the WITH expression.
 
-## Limitations {#limitations}
+## Syntax
 
-1.  Recursive queries are not supported.
-2.  When subquery is used inside WITH section, it’s result should be scalar with exactly one row.
-3.  Expression’s results are not available in subqueries.
+``` sql
+WITH <expression> AS <identifier|expression>
```

Comment:
AS expression? Is it true?

Reply:
I feel that we should document it like this
Nullable indices and index usage

```
WITH <expression> AS <identifier>
|
WITH <identifier> AS <subquery expression>
```

When we encounter an subquery after `AS`, the second case takes precedence, or else the first one wins.